Source:   svn r1167826 (4.4 branch), svn r1167827 (4.5 branch)
Upstream: Yes
Reason:   Okular PDB Processing Memory Corruption Vulnerability
          KDE Security Advisory 2010-08-25-1 (CVE-2010-2575; SA40952)
          http://www.kde.org/info/security/advisory-20100825-1.txt

--- kdegraphics/okular/generators/plucker/unpluck/image.cpp	2010/08/25 12:32:46	1167825
+++ kdegraphics/okular/generators/plucker/unpluck/image.cpp	2010/08/25 12:32:55	1167826
@@ -289,8 +289,23 @@
             for (j = 0; j < bytes_per_row;) {
                 incount = *palm_ptr++;
                 inval = *palm_ptr++;
-                memset (rowbuf + j, inval, incount);
-                j += incount;
+                if (incount + j <= bytes_per_row  * width)
+                {
+                    memset (rowbuf + j, inval, incount);
+                    j += incount;
+                }
+                else
+                {
+                    free (rowbuf);
+                    free (lastrow);
+                    free (jpeg_row);
+
+                    jpeg_destroy_compress (&cinfo);
+
+                    fclose( outfile );
+
+                    return false;
+                }
             }
         }
         else if ((flags & PALM_IS_COMPRESSED_FLAG)
